CMakeLists.txt

```diff
@@ -61,6 +61,28 @@ add_subdirectory(drivers)
 ################################################################################
 # Testing                                                                      #
 ################################################################################
-enable_testing()
+option(TEST_MODE "Set what validation target the tests should be run on" "OFF")
+set(SIMULATION_BINARY "chim.bin" CACHE STRING "Path to the simulation binary")
+set(PRELOAD_MODE "0" CACHE STRING "Preload mode for simulation")
 
-add_subdirectory(tests)
+if (TEST_MODE STREQUAL "simulation")
+  enable_testing()
+
+  if(NOT EXISTS ${SIMULATION_BINARY})
+    message(FATAL_ERROR "Simulation binary ${SIMULATION_BINARY} does not exist.")
+  endif()
+
+  # Convert PRELOAD_MODE to integer
+  if ("${PRELOAD_MODE}" MATCHES "^[0-9]+$")
+    set(PRELOAD_MODE_INT ${PRELOAD_MODE})
+  else()
+    message(FATAL_ERROR "PRELOAD_MODE must be an integer.")
+  endif()
+
+  message(STATUS "[CHIMERA-SDK] Simulation Binary     : ${SIMULATION_BINARY}")
```

Comment:
"Simulation Binary" could be confused with the binary to be simulated. Why not name it "Simulation Platform"?

Reply:
See previous comment - it refers to the binary that is created by the simulation tool to model the SoC's RTL; it's not a platform, as it's not supposed to include things like GVSoC.

Reply:
Maybe we could rename it to something like SoC Model Binary? Would that be clearer?

Reply:
Ok, I see. Yes, let's rename it to `SoC Model Binary` and the variable to `SOC_MODEL_BINARY`.

Reply:
She's renamed :)